ivy/array/general.py

```diff
@@ -173,6 +173,44 @@ def gather_nd(
         """
         return ivy.gather_nd(self, indices, out=out)
 
+    def scatter_nd(self: ivy.Array, 
+        indices:Union[ivy.Array, ivy.NativeArray], 
+        updates:Union[ivy.Array, ivy.NativeArray],
+        shape: Union[ivy.Array, ivy.NativeArray] = None,
+        reduction: str = "sum",
+        *,
+        out: Optional[ivy.Array] = None,
+        ) -> ivy.Array:
+        """
+        Scatter updates into an array according to indices.
+
+        Parameters
+        ----------
+        self
+            The tensor in which to scatter the results
+        indices
+            Tensor of indices
+        updates
+            values to update input tensor with
+
+        Returns
+        -------
+        ret
+            New array of given shape, with the values scattered at the indices.
+
+        Examples
+        --------
+        scatter values into an array
```

Comment:
Please use correct input type

Reply:
what is the correct input type? Should I omit ivy.NativeArray?

Reply:
By that, I meant to specify the input instance type, e.g. "With :code:`ivy.Array` input:" could be used with `ivy.`Array instances in the input. Going through this would help: https://github.com/unifyai/ivy/blob/master/docs/partial_source/deep_dive/13_docstring_examples.rst

Reply:
The checks keep failing. The errors are in files I haven't modified. What can I do about that?

Reply:
Other errors are just related to functions beyond the current issue e.g. 'current_backend_str', 'test_sparse_categorical_crossentropy', etc.

Reply:
If failing tests have nothing to do with this commit then we will merge this commit. No worries about that :)